src/test/java/com/endava/internship/mocking/repository/InMemPaymentRepositoryTest.java

```diff
@@ -0,0 +1,73 @@
+package com.endava.internship.mocking.repository;
+
+import com.endava.internship.mocking.model.Payment;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.*;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.assertj.core.api.Assertions.*;
+
+class InMemPaymentRepositoryTest {
+    InMemPaymentRepository inMemPaymentRepository;
+    Payment p1;
+    Payment p2;
+    Payment p3;
+
+    @BeforeEach
+    void setUp() {
+        inMemPaymentRepository = new InMemPaymentRepository();
+        p1 = new Payment(1,100d,"message1");
+        p2 = new Payment(2,200d,"message2");
+        p3 = new Payment(3,300d,"message3");
+        inMemPaymentRepository.save(p1);
+        inMemPaymentRepository.save(p2);
+        inMemPaymentRepository.save(p3);
+    }
+
```

Comment:
Add @AfterEach method and verifyNoMoreInteractions for every mock

Reply:
fixed